app/src/main/java/protect/card_locker/preferences/SettingsActivity.java

```diff
@@ -109,7 +112,7 @@ public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
             });
             localePreference.setOnPreferenceChangeListener((preference, newValue) -> {
                 // Refresh the activity
-                Intent intent = new Intent(getContext(), SettingsActivity.class);
+                Intent intent = new Intent(getContext(), MainActivity.class);
```

Comment:
I think we need to add something to MainActivity that will open another activity, so we can go back to settings while force-reloading main too. That way we can also fix #607 in the same go.